src/coreComponents/mesh/SurfaceElementRegion.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SurfaceElementRegion.cpp
+ */
+
+#include "EdgeManager.hpp"
+#include "SurfaceElementRegion.hpp"
+
+
+namespace geosx
+{
+using namespace dataRepository;
+
+SurfaceElementRegion::SurfaceElementRegion( string const & name, Group * const parent ):
+  ElementRegionBase( name, parent )
+{
+  registerWrapper( viewKeyStruct::defaultApertureString, &m_defaultAperture )->
+    setInputFlag( InputFlags::REQUIRED )->
+    setDescription( "The default aperture of newly formed surface elements." );
+}
+
+SurfaceElementRegion::~SurfaceElementRegion()
+{}
+
+
+void SurfaceElementRegion::GenerateMesh( Group * const cellBlocks )
+{
+  GEOSX_UNUSED_VAR( cellBlocks );
+
+  Group * const elementSubRegions = this->GetGroup( viewKeyStruct::elementSubRegions );
+
+  elementSubRegions->RegisterGroup< EmbeddedSurfaceSubRegion >( "embeddedSurfaceSubRegion" );
+
+  elementSubRegions->RegisterGroup< FaceElementSubRegion >( "faceElementSubRegion" );
```

Comment:
@rrsettgast @joshua-white I have tried to have both subregions and just fill them accordingly. This generates runtime errors though whenever one of them is basically empty. For now, I don't really see a solution apart from specifying the subRegionType in the input.